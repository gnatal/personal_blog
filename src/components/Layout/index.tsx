import React from 'react';
import NavBar from '../NavBar';

export default function Layout({ children }: {children: any}) {
  return (
    <>
      <NavBar blogName="Blog do Natal" />
      <main>
        {children}
      </main>

    </>
  );
}
