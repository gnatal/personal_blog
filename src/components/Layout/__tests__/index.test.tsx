import React from 'react';
import renderer from 'react-test-renderer';
import Layout from '..';

test('It should show the nav bar and a simple text', () => {
  const component = renderer.create(
    <Layout>
      <p> just a text</p>
    </Layout>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
