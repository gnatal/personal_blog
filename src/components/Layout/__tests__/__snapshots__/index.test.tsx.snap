// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should show the nav bar and a simple text 1`] = `
Array [
  <nav
    className="flex items-center justify-between flex-wrap bg-teal-500 p-6"
  >
    <div
      className="flex items-center flex-shrink-0 text-white mr-6"
    >
      <span
        className="font-semibold text-xl tracking-tight"
      >
        Blog do Natal
      </span>
    </div>
    <div
      className="w-full hidden flex-grow sm:flex sm:items-center sm:w-auto sm:block"
    >
      <div
        className="text-sm sm:flex-grow"
      >
        <a
          className="block mt-4 md:mt-2 text-md sm:inline-block lg:mt-0 text-teal-100 hover:text-white mr-4"
          href="/"
        >
          início
        </a>
        <a
          className="block mt-4 md:mt-2 text-md sm:inline-block lg:mt-0 text-teal-100 hover:text-white mr-4"
          href="/about"
        >
          sobre
        </a>
        <a
          className="block mt-4 md:mt-2 text-md sm:inline-block lg:mt-0 text-teal-100 hover:text-white mr-4"
          href="/posts"
        >
          blog
        </a>
      </div>
      <div>
        <a
          className="inline-block text-sm px-4 py-2 leading-none border rounded text-white border-white hover:border-transparent hover:text-teal-500 hover:bg-white mt-4 md:mt-0"
          href="/login"
        >
          login
        </a>
      </div>
    </div>
    <div
      className="block sm:hidden"
    >
      <button
        className="flex items-center px-3 py-2 border rounded text-teal-200 border-teal-400 hover:text-white hover:border-white"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="fill-current h-3 w-3"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z"
          />
        </svg>
      </button>
    </div>
  </nav>,
  <div
    className="block sm:hidden bg-teal-500 h-screen w-screen"
  >
    <div
      className="w-screen pl-6 pb-2 pt-2 border"
    >
      <a
        className="block font-bold text-md text-teal-100 hover:text-white"
        href="/"
      >
        início
      </a>
    </div>
    <div
      className="w-screen pl-6 pb-2 pt-2 border"
    >
      <a
        className="block font-bold text-md text-teal-100 hover:text-white"
        href="/about"
      >
        sobre
      </a>
    </div>
    <div
      className="w-screen pl-6 pb-2 pt-2 border"
    >
      <a
        className="block font-bold text-md text-teal-100 hover:text-white"
        href="/posts"
      >
        blog
      </a>
    </div>
    <div
      className="w-screen pl-6 pb-2 pt-2 border"
    >
      <a
        className="block font-bold text-md text-teal-100 hover:text-white"
        href="/login"
      >
        login
      </a>
    </div>
  </div>,
  <main>
    <p>
       just a text
    </p>
  </main>,
]
`;
